src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1080,6 +1087,22 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
       case Token.NUMBER:
         if (right.isNumber()) {
           return compareAsNumbers(op, left, right, useTypes);
+        } else {
+          // See http://www.ecma-international.org/ecma-262/6.0/#sec-abstract-relational-comparison
+          switch (op) {
+            case Token.GE:
+            case Token.LE:
+            case Token.GT:
+            case Token.LT:
+              return compareAsNumbers(op, left, right, useTypes);
+            case Token.EQ:
+            case Token.NE:
+              // TODO(moz): We can fold more cases, but testCommutativeOperators() will fail
```

Comment:
I'm not sure I follow this TODO. If we did this for non-strings, it would be wrong.
e.g.
0 == null would be folded to true instead of false.